zkp-circuits/README.md

```diff
@@ -0,0 +1,100 @@
+# Nimiq ZKP Circuits
+This crate contains the zero-knowledge proof circuits for Nimiq's light client.
+Together with the code in the `nimiq-zkp` and `nimiq-zkp-primitives` crates, it allows to recursively prove the state of the chain.
+
+Our circuits are built on top of [arkworks](https://arkworks.rs).
+We perform Groth16 SNARK proofs on the MNT4/6 elliptic curve cycle (specifically the MNT4-753 and MNT6-753 curves).
+We can use the MNT4 curve to prove statements in the MNT6 curve and vice-versa (you can find an in-depth introduction to recursive SNARKs [here](https://www.michaelstraka.com/posts/recursivesnarks/)).
+
+More detailed information on Nimiq's blockchain protocol can be found [here](https://arxiv.org/abs/1903.01589).
+
+## Crate Structure
+This crate is structured as follows:
+
+- `gadgets`: Gadgets are smaller building blocks containing intermediate computations that can be composed to build a more complete statement. This folder contains curve-independent gadgets, while curve-specific gadgets are placed in the corresponding subfolders.
+- `circuits`: Contains the ZKP circuits (more details below). A circuit may combine multiple gadgets to prove a larger statement. Our circuits are always specific to a curve.
+
+Each of these folders contains two folders `mnt4` and `mnt6` for curve-specific code.
+Note that this refers to the curve the statements are made in and not the curve the proof will be generated with.
+A circuit in the `mnt6` folder, for example, will contain statements on the MNT6-753 curve.
+We will prove these statements using Groth16 over MNT4-753.
+
+## Circuits
+At a very high level, we use recursive SNARKs to compress the work of verifying many blocks into a single short proof.
+Our setup also allows us to update a proof when a new block comes in, without having knowledge of the past blocks.
+We only need the previous proof and the new block.
+
+This library consists of several circuits that, when chained together, can recursively prove the validity of many blocks.
+The image below illustrates that: 
+
+![Light Node Setup](images/light_node.png)
+
+The arrows represent that the proof generated by one circuit is given as input to another circuit.
+The given curve indicates in which folder it can be found.
+A brief description of each circuit:
+
+* __Macro Block Circuit__: This circuit verifies the validity of a single block.
+* __Macro Block Wrapper Circuit__: This circuit verifies a SNARK proof of the Macro Block circuit. It is used to _wrap_ the Macro Block proof so that it can be fed into the Merger circuit.
+* __Merger Circuit__: This circuit verifies two SNARK proofs. It is used to _merge_ two different proofs into a single one.
+* __Merger Wrapper Circuit__: This circuit verifies a SNARK proof of the Merger circuit. It is used
+to _wrap_ the Merger proof so that it can be fed into the Merger circuit.
```

Comment:
```suggestion
* __Merger Wrapper Circuit__: This circuit verifies a SNARK proof of the Merger circuit. It is used to _wrap_ the Merger proof so that it can be fed into the Merger circuit.
```